test/integration/sequelize-custom.test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const { expect } = require('chai');
+const Sequelize = require('../../index');
+const config = require('../config/config');
+
+const dialectName = process.env.DIALECT === 'postgres-native' ? 'postgres' : process.env.DIALECT;
+const CustomDialect = require(`../../lib/dialects/${dialectName}`);
+
+before(() => {
+  const options = {
+    ...config[dialectName],
+    dialect: CustomDialect
+  };
+  
+  this.sequelize = new Sequelize(options);
+});
+
+after(() => {
+  this.sequelize.close();
+});
+
+describe('Custom Dialect', () => {
+  describe('authenticate', () => {
+    it('connects to database successfully', async () => {
+      try {
+        await this.sequelize.authenticate();
+        expect(true).to.equal(true);
+      } catch (error) {
+        console.error(error);
+      }
```

Comment:
Using try/catch in tests is not safe. If this promise rejects the test suite will not fail. I see we have [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/) installed in the package, so we can do:

```suggestion
       return  expect(this.sequelize.authenticate()).to.eventually.equal(true);
```